Kernel/FileSystem/BlockBasedFileSystem.cpp

```diff
@@ -60,6 +60,10 @@ class DiskCache {
             return nullptr;
         auto& entry = const_cast<CacheEntry&>(*it->value);
         VERIFY(entry.block_index == block_index);
+        if (!entry_is_dirty(entry)) {
+            // Cache hit! Promote the entry to the front of the list.
+            m_clean_list.prepend(entry);
```

Comment:
it seems you have `mark_clean`… also not sure whether you should switch to that… 🤔

Reply:
Even though `mark_clean` does the same operation of prepending the entry, it is not what we are doing here. We are moving the entry that was already clean because it is more likely to be touched again.